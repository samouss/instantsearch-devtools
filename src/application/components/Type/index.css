.Type {
  padding: 4px 8px;
  font-weight: 600;
  color: white;
  font-size: 0.85rem;
}

.Type--change {
  background-color: #5560B5;
}

.Type--search {
  background-color: #008FBA;
}

.Type--result {
  background-color: #3AB2BD;
}
