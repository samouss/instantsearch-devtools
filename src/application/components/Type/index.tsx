import React, { SFC } from 'react';
import cx from 'classnames';
import styles from './index.css';

type Props = {
  name: string;
};

const Type: SFC<Props> = ({ name }) => (
  <span className={cx(styles.Type, styles[`Type--${name.toLowerCase()}`])}>
    {name.toUpperCase()}
  </span>
);

export default Type;
