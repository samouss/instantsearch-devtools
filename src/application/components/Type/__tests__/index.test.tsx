import React from 'react';
import { shallow } from 'enzyme';
import Type from '../index';

describe('<Type />', () => {
  it('expect to render with "CHANGE"', () => {
    const props = {
      name: 'CHANGE',
    };

    const wrapper = shallow(<Type {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with "SEARCH"', () => {
    const props = {
      name: 'SEARCH',
    };

    const wrapper = shallow(<Type {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with "RESULT"', () => {
    const props = {
      name: 'RESULT',
    };

    const wrapper = shallow(<Type {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to be case insensitive', () => {
    const props = {
      name: 'reSuLt',
    };

    const wrapper = shallow(<Type {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
