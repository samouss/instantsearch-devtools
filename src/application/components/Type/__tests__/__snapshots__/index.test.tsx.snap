// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Type /> expect to be case insensitive 1`] = `
<span
  className="Type Type--result"
>
  RESULT
</span>
`;

exports[`<Type /> expect to render with "CHANGE" 1`] = `
<span
  className="Type Type--change"
>
  CHANGE
</span>
`;

exports[`<Type /> expect to render with "RESULT" 1`] = `
<span
  className="Type Type--result"
>
  RESULT
</span>
`;

exports[`<Type /> expect to render with "SEARCH" 1`] = `
<span
  className="Type Type--search"
>
  SEARCH
</span>
`;
